client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -1,544 +1,659 @@
 }) => {
+  const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
   const recordBatchResults = useRecordBatchResults(
     electionId,
     jurisdictionId,
     roundId
   )
-  const initialTallySheets =
-    batch.resultTallySheets.length > 1
-      ? batch.resultTallySheets
-      : batch.resultTallySheets.concat([
-          emptyTallySheet(batch.resultTallySheets.length + 1),
-        ])
-  const { register, control, handleSubmit, formState, errors } = useForm<{
-    resultTallySheets: IBatchResultTallySheet[]
-  }>({
-    defaultValues: { resultTallySheets: initialTallySheets },
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [selectedTabId, setSelectedTabId] = useState('')
+  const [isEditing, setIsEditing] = useState(false)
+  const [isRenamingSheet, setIsRenamingSheet] = useState(false)
+  const [sheetNameBeforeRename, setSheetNameBeforeRename] = useState<{
+    index: number
+    name: string
+  }>()
+
+  const batches = batchesQuery.isSuccess ? batchesQuery.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  const initialResultTallySheets = selectedBatch?.resultTallySheets.length
+    ? selectedBatch.resultTallySheets
+    : [constructEmptyResultTallySheet(0)]
+  const formMethods = useForm<IBatchRoundDataEntryFormState>({
+    defaultValues: {
+      resultTallySheets: initialResultTallySheets,
+    },
   })
-  const { fields, append, remove } = useFieldArray({
+  const {
+    control,
+    formState,
+    getValues,
+    handleSubmit,
+    reset,
+    setValue,
+    watch,
+  } = formMethods
+  const {
+    append: appendSheet,
+    fields: sheets,
+    remove: removeSheet,
+  } = useFieldArray<IBatchResultTallySheet>({
     control,
     name: 'resultTallySheets',
   })
 
-  const onSubmit = async ({
+  // Reset state whenever a new batch is selected
+  useEffect(() => {
+    reset({ resultTallySheets: initialResultTallySheets })
+    setSelectedTabId(VOTE_TOTALS_TAB_ID)
+    setIsEditing(false)
+  }, [selectedBatchId])
+
+  // Auto-select first batch on initial load
+  useEffect(() => {
+    if (!selectedBatchId && filteredBatches.length > 0) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [filteredBatches, selectedBatchId, setSelectedBatchId])
+
+  // Auto-select first search match
+  useEffect(() => {
+    if (
+      debouncedSearchQuery &&
+      filteredBatches.length > 0 &&
+      !formState.isDirty
+    ) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [debouncedSearchQuery, filteredBatches, formState, setSelectedBatchId])
```

Comment:
I wonder if we even need the `selectedBatchId` state anymore. Could it just be derived as the first item in `filteredBatches`, since we always select the first item in the list now?

Reply:
Ah yes good q - we do still need it for the case that someone explicitly clicks on a batch from the list, which of course won't always be the first batch